src/App.tsx

```diff
@@ -23,21 +23,20 @@ var todos = (): Promise<{id: string; title: string;}[]> => new Promise((res) =>
 });
 
 function App() {
-  const [state, setState] = React.useState<{ id: string; title: string }[]>([]);
+  const [todoList, setTodoList] = React.useState<{ id: string; title: string }[]>([]);
 
   React.useEffect(() => {
     (async () => {
-      var awaitedTodos = await todos();
-      for (var i = 0; i < awaitedTodos.length; i++) {
-        setState([...state, awaitedTodos[i]]);
-      }
+      const awaitedTodos = await getTodosService();
+      setTodoList(awaitedTodos);
     })()
-  })
+  },[])
+
 
   return (
     <div>
-      {state.map((todo) => (
-        <Todo todo={todo} />
```

Comment:
The Key is for the identification of items in the array. 
If you don't provide keys, React will not know which items have been changed in the array then all nodes will be rerendered.
Keys help React identify which items have changed, are added, or are removed. But don't use an index as a key. If you provide a key, React will skip unchanged nodes.